File: problems/two-sum.go
package main

func twoSum(nums []int, target int) []int {
	complementMap := make(map[int]int)

	for i, num := range nums {
		diff := target - num

		if idx, exists := complementMap[num]; exists {
			return []int{idx, i}
		}

		complementMap[diff] = i
	}

	return []int{}
}

File: problems/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map();

    for (let i = 0; i < nums.length; i++) {
      const diff = target - nums[i];
      const a = map.get(diff);

      if (a != undefined) 
        return [a,i];
      }
      

      map.set(nums[i], i);
    }
};
